pipeline.ml

```diff
@@ -28,86 +28,110 @@ let dockerfile ~base =
+      | `PR num ->
+          let dockerfile =
+            let+ base = Docker.pull ~schedule:weekly "ocaml/opam2" in
+            `Contents (dockerfile ~base)
+          in
+          let docker_cpuset_cpus =
+            match docker_cpu with
+            | Some i -> [ "--cpuset-cpus"; string_of_int i ]
+            | None -> []
+          in
+          let docker_cpuset_mems =
+            match docker_numa_node with
+            | Some i -> [ "--cpuset-mems"; string_of_int i ]
+            | None -> []
+          in
+          let tmpfs =
+            match docker_numa_node with
+            | Some i ->
+                [
+                  "--tmpfs";
+                  Fmt.str "/dev/shm:rw,noexec,nosuid,size=%dG,mpol=bind:%d"
+                    docker_shm_size i;
+                ]
+            | None ->
+                [
+                  "--tmpfs";
+                  Fmt.str "/dev/shm:rw,noexec,nosuid,size=%dG" docker_shm_size;
+                ]
+          in
+          let s =
+            let run_args =
+              [ "--security-opt"; "seccomp=./aslr_seccomp.json" ]
+              @ tmpfs
+              @ docker_cpuset_cpus
+              @ docker_cpuset_mems
+            in
+            let+ output =
+              let src =
+                Git.fetch
+                  (Current.map Github.Api.Commit.id (Current.return head))
+              in
+              let image =
+                Docker.build ~pool ~pull:false ~dockerfile (`Git src)
+              in
+              Docker.pread ~run_args image
+                ~args:
+                  [
+                    "/usr/bin/setarch";
+                    "x86_64";
+                    "--addr-no-randomize";
+                    "_build/default/bench/bench.exe";
+                    "--nb-entries";
+                    "1000";
+                    "-d";
+                    "/dev/shm";
+                    "--json";
+                  ]
+            in
+            let commit = Github.Api.Commit.hash head in
+            let content =
+              Utils.merge_json name commit (Yojson.Basic.from_string output)
+            in
+            let () = Utils.populate_postgres conninfo commit content num in
+            match slack_path with Some p -> Some (p, content) | None -> None
+          in
+          s
+          |> Current.option_map (fun p ->
+                 Current.component "post"
+                 |> let** path, _ = p in
+                    let channel = read_channel_uri path in
+                    Slack.post channel ~key:"output" (Current.map snd p))
+          |> Current.ignore_value)
+    refs
+
+let pipeline ~github ~(repo : Github.Repo_id.t) ?slack_path ?docker_cpu
+    ?docker_numa_node ~docker_shm_size ~conninfo () =
+  let name = repo.name in
+  let repo = Current.return (github, repo) in
+  process_pipeline ~docker_cpu ~docker_numa_node ~docker_shm_size ~conninfo ~name ~repo
+    ~slack_path
   |> Current.ignore_value
```

Comment:
you can also remove this line, by ensuring that `process_pipeline` return `unit Current.t`. I think it should be feasible as ` Slack.post` returns `unit Current.t` and that's the last function you call.